### PR TITLE
Mobile builds: expose main sub-servers by default

### PR DESCRIPTION
**Pull request by kaloudis**
## Change Description
Update the mobile build system to also properly expose all the sub-severs in a proper namespace.

Previously, only `autopilotrpc` was exposed.

Now we'll also expose:
- `chainrpc`
- `invoicesrpc`
- `peersrpc`
- `routerrpc`
- `signrpc`
- `verrpc`
- `walletrpc`
- `watchtowerrpc`
- `wtclientrpc`

TODO

- [x] make sure the stubs show up properly in Xcode/Android Studio
- [x] assess new build sizes

## Steps to Test
Steps for reviewers to follow to test the change.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [ ]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

**Comment by kaloudis** [comment]
@Roasbeef any reason we shouldn't put `SUBSERVER_PREFIX` on by default in the `mobile-rpc` step? We certainly get conflicts.

**Comment by kaloudis** [comment]
Output on `make android` is roughly 1.2% bigger (49.4 MB -> 50 MB)

Output on `make ios` is roughly 2.2% bigger (133.9 MB -> 136.9 MB)

**Comment by guggero** [comment]
> @Roasbeef any reason we shouldn't put `SUBSERVER_PREFIX` on by default in the `mobile-rpc` step? We certainly get conflicts.

I think at this point it's mostly a question of remaining backward compatible with previous builds. But I assume because you'd get conflicts when running without `prefix=1` everyone is already using that anyway? I checked Blixt wallet and it seems to already prefix.
So perhaps we should turn that on by default in 0.15?

**Comment by Roasbeef** [comment]
So seems we didn't actually need something similar to this in the main bash script https://github.com/lightningnetwork/lnd/issues/6296#issuecomment-1055943017 ?

**Comment by kaloudis** [comment]
@Roasbeef I see the header bindings for all the calls added to the mobile binaries. could setting the listener to all the services still be required?

**Comment by Roasbeef** [comment]
>  I see the header bindings for all the calls added to the mobile binaries. could setting the listener to all the services still be required?

Have we confirmed the calls can actually be executed? AFAIU, that's required since each exposed gRPC service needs to listen on a unique bufconn object.

**Comment by Roasbeef** [comment]
Ping on this re testing e2e (at least one call for each sub-server).

File: Makefile [hunk 1]
```diff
@@ -293,8 +293,8 @@ sample-conf-check:
 	for flag in $$(GO_FLAGS_COMPLETION=1 go run -tags="$(RELEASE_TAGS)" $(PKG)/cmd/lnd -- | grep -v help | cut -c3-); do if ! grep -q $$flag sample-lnd.conf; then echo "Command line flag --$$flag not added to sample-lnd.conf"; exit 1; fi; done
 
 mobile-rpc:
-	@$(call print, "Creating mobile RPC from protos (prefix=$(prefix)).")
-	cd ./lnrpc; COMPILE_MOBILE=1 SUBSERVER_PREFIX=$(prefix) ./gen_protos_docker.sh
+	@$(call print, "Creating mobile RPC from protos.")
```
Comment (Roasbeef) [inline Makefile:296]: We should update the docs that his is no longer needed. 

Does this change the API for the existing apps using the non-prefix version?

File: mobile/gen_bindings.sh [hunk 2]
```diff
@@ -33,13 +33,13 @@ target_pkg="github.com/lightningnetwork/lnd/lnrpc"
 
 # A mapping from grpc service to name of the custom listeners. The grpc server
 # must be configured to listen on these.
-listeners="lightning=lightningLis walletunlocker=lightningLis state=lightningLis"
+listeners="lightning=lightningLis walletunlocker=lightningLis state=lightningLis autopilot=lightningLis chainnotifier=lightningLis invoices=lightningLis neutrinokit=lightningLis peers=lightningLis router=lightningLis signer=lightningLis walletkit=lightningLis watchtower=lightningLis watchtowerclient=lightningLis"
```
Comment (Roasbeef) [inline mobile/gen_bindings.sh:36]: We should also update the docs to explain what the new API object people will need to hit to access the extra sub-severs looks like.

**Review by Roasbeef — CHANGES_REQUESTED** [review]

**Review by Roasbeef — APPROVED** [review]
LGTM 🦚

Final thing this is missing is an update to teh release notes.

File: Makefile [hunk 3]
```diff
@@ -29,11 +29,13 @@ COMMIT_HASH := $(shell git rev-parse HEAD)
 
 GOBUILD := go build -v
 GOINSTALL := go install -v
-GOTEST := go test 
+GOTEST := go test
 
 GOVERSION := $(shell go version | awk '{print $$3}')
 GOFILES_NOVENDOR = $(shell find . -type f -name '*.go' -not -path "./vendor/*" -not -name "*pb.go" -not -name "*pb.gw.go" -not -name "*.pb.json.go")
 
+SUBSERVERS := autopilotrpc chainrpc invoicesrpc neutrinorpc peersrpc routerrpc signrpc walletrpc watchtowerrpc wtclientrpc
```
Comment (guggero) [inline Makefile:37]: This is already defined as `RPC_TAGS` in `make/testing_flags.mk`. I think we should re-use that variable instead, even if it defines build tags that aren't actually build tags (`routerrpc` and `verrpc` aren't behind build tags, they're always built).

Comment (guggero) [reply Makefile:37]: Ah, but I see that the falafel script adds build tags to the generated mobile files, so we actually need the additional `routerrpc` and `verrpc` tags from the `RPC_TAG` variable!

**Review by guggero — APPROVED** [review]
LGTM, just one minor correction is needed.

File: Makefile (hunk 3 above)
Comment (kaloudis) [reply Makefile]: `routerrpc` was already there, but I've removed the `SUBSERVERS` var in favor of `RPC_TAGS`. I've also added `verrpc` to the falafel script.

File: Makefile (hunk 1 above)
Comment (guggero) [reply Makefile:294]: This does change existing apps, yes. At least if they use the non-prefixed version. But given that there are collisions if you don't use the prefix, maybe nobody was able to use the non-prefixed code for a while and there is nobody affected?

Comment (kaloudis) [reply Makefile:294]: Thinking of the most used LND mobile apps right now:

- Breez: forked, has its own build system
- Blixt: already using prefix

Any others come to mind we should check out?

Comment (guggero) [reply Makefile:294]: We could post a question in the #mobile channel on Slack. Maybe we'll get some more insight there?

Comment (kaloudis) [reply Makefile:294]: Good idea

Comment (guggero) [reply Makefile:294]: I'm going to merge this in the meantime. We can always roll this back during the RC phase if anyone has issues with the prefix.

**Pull request merged**